carma_wm/src/MapConformer.cpp

```diff
@@ -480,10 +478,79 @@ void addInferredDirectionOfTravel(Lanelet& lanelet, lanelet::LaneletMapPtr map,
   }
 }
 
+void addValidSpeedLimit(Lanelet& lanelet, lanelet::LaneletMapPtr map, lanelet::Velocity config_limit,
+    const std::vector<lanelet::traffic_rules::TrafficRulesUPtr>& default_traffic_rules )
+{
+  lanelet::Velocity max_speed;
+    auto speed_limit = lanelet.regulatoryElementsAs<DigitalSpeedLimit>();
+     if(config_limit < 80_mph && config_limit > 0_mph)//Accounting for the configured speed limit, input zero when not in use
+      {  
+        max_speed = config_limit;
+      }
+      else
+      {
+        max_speed = 80_mph;
+      }
+      
+    // If the lanelet does not have a digital speed limit then add one with the maximum value of 80
+    std::vector<std::string> allowed_participants;
+     //Maximum speed limit is 80
+   
+
+    if (speed_limit.size()== 0)//If there is no assigned speed limit value
+    {
+              ROS_WARN_STREAM(" check");
+
+         for(const auto& rules : default_traffic_rules)
+         {
+            if (rules->canPass(lanelet))
+             {
+              allowed_participants.emplace_back(rules->participant());
+             }
+        }
+     if (allowed_participants.size() > 0)
+     {
+      std::shared_ptr<DigitalSpeedLimit> rar(new DigitalSpeedLimit(DigitalSpeedLimit::buildData(lanelet::utils::getId(), max_speed, {lanelet},
+      {}, allowed_participants))); 
+
+      lanelet.addRegulatoryElement(rar);
+      map->add(rar);//Add DigitalSpeedLimit data to the map
+
+     }
+  }
+  else  //If the speed limit value already exists 
+  {
+    ROS_WARN_STREAM("check3");
+      for(const auto& rules : default_traffic_rules)
+         {
+            if (rules->canPass(lanelet))
+             {
+              allowed_participants.emplace_back(rules->participant());
+             }
+        }
+    if(speed_limit.back().get()->speed_limit_ > max_speed)//Check that speed limit value does not exceed the maximum value
+    {
+      ROS_WARN_STREAM("Invalid speed limit value. Value reset to maximum speed limit.");
+      std::shared_ptr<DigitalSpeedLimit> rar(new DigitalSpeedLimit(DigitalSpeedLimit::buildData(lanelet::utils::getId(), max_speed, {lanelet},
+      {}, allowed_participants)));
+      //map->remove(lanelet, speed_limit.back());
```

Comment:
if using this works, pls use this and delete the line below. vice versa.
just have one of them, and no code in commented out